### PR TITLE
SDK-389: Break up SecureYotiClient

### PR DESCRIPTION
**Pull request by bucky-boy**
Some big changes, but there should be no change to the API or the behaviour of the SDK to anyone using it.

More detail in the ticket.

### CODE REVIEW

File: yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/ActivityDetailsFactory.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import javax.crypto.spec.SecretKeySpec;
+
+import com.yoti.api.client.ActivityDetails;
+import com.yoti.api.client.Profile;
+import com.yoti.api.client.ProfileException;
+import com.yoti.api.client.spi.remote.call.Receipt;
+import com.yoti.api.client.spi.remote.util.Decryption;
+
+public class ActivityDetailsFactory {
```
Comment (qzagarese) [inline yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/ActivityDetailsFactory.java]: Why is this Factory public? 
It really shouldn't be used from client code.

File: yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/util/Decryption.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import static com.yoti.api.client.spi.remote.call.YotiConstants.BOUNCY_CASTLE_PROVIDER;
+
+import java.security.GeneralSecurityException;
+import java.security.PrivateKey;
+
+import javax.crypto.Cipher;
+
+import com.yoti.api.client.ProfileException;
+
+public class Decryption {
```
Comment (qzagarese) [inline yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/util/Decryption.java]: can we come up with a different kind of naming? `CryptoHelper` or something along those lines?

File: yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/ActivityDetailsFactory.java (hunk 1 above)
Comment (bucky-boy) [reply yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/ActivityDetailsFactory.java]: Fixed that one, and a few others where public access is not required.

This does force us to keep components in the same package, which is not ideal, but if we want to restrict access I don't see another way of doing it.

File: yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/util/Decryption.java (hunk 2 above)
Comment (bucky-boy) [reply yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/util/Decryption.java]: It's now named DecryptionHelper.

**Pull request merged**